### PR TITLE
feat: improve DynamicConfigForm

### PR DESCRIPTION
**Pull request by ryanio**
* Displays current config value from store, instead of default value from settings
* Adds `path` form item type
* Creates `DynamicConfigFormItem.jsx` to simplify parent component
* Removes old hardcoded geth `ConfigForm.jsx`

Path type settings field:
Screen Shot 2019-05-16 at 9 23 47 AM

### CODE REVIEW

File: src/components/Nodes/ClientConfig/DynamicConfigForm/FormItem.jsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+  render() {
+    const { itemKey, item, client, clientName, isClientRunning } = this.props
+    const label = item.label || itemKey
+    let { type } = item
+    if (!type) type = item.options ? 'select' : 'text'
+    const itemValue = client[clientName].config[itemKey]
+    let options
+
+    switch (type) {
```
Comment (marcgarreau) [inline src/components/Nodes/ClientConfig/DynamicConfigForm/FormItem.jsx]: Nit: each type might be a good candidate for its own `shared` component. Can take place in a future PR.

File: src/components/Nodes/ClientConfig/index.js [hunk 2]
```diff
@@ -7,7 +7,6 @@ import Tabs from '@material-ui/core/Tabs'
 import Tab from '@material-ui/core/Tab'
 import Typography from '@material-ui/core/Typography'
 import VersionList from './VersionList'
-// import ConfigForm from './ConfigForm'
```
Comment (marcgarreau) [inline src/components/Nodes/ClientConfig/index.js]: Should just be deleted.

File: src/components/Nodes/ClientConfig/DynamicConfigForm/FormItem.jsx (hunk 1 above)
Comment (ryanio) [reply src/components/Nodes/ClientConfig/DynamicConfigForm/FormItem.jsx]: yes agreed, was tempted to break it out but perhaps on adding the next type when it gets too long

File: src/components/Nodes/ClientConfig/index.js (hunk 2 above)
Comment (ryanio) [reply src/components/Nodes/ClientConfig/index.js]: it is deleted :)

**Comment by ryanio** [comment]
> Note: on the Grid side, Parity config file needs updating ('path' type, etc.).

thanks for the catch, added here https://github.com/ethereum/grid/commit/10098554df77efd40420c55cd7594c0f2883a3f9

**Pull request merged**